R-packages/evalcast/R/err_measures.R

```diff
@@ -24,75 +22,100 @@ weighted_interval_score <- function(quantile_forecasts, actual_value) {
   #
   # (|y - median| + sum_k [alpha_k*width_k/2 + dfi_k]) / (num_intervals + 1)
   # where dfi_k = dist_from_interval_k
-  if (is.na(actual_value)) return(NA)
-  num_prob <- nrow(quantile_forecasts) # 23
-  assert_that(num_prob %% 2 == 1 && num_prob >= 3,
-              msg=paste("Number of predicted quantiles must be an odd number",
-                        "greater than or equal to 3."))
+  point_fcast <- is.na(quantile)
+  quantile <- quantile[!point_fcast]
+  value <- value[!point_fcast]
+  if (all(is.na(actual_value))) return(NA)
+  actual_value <- actual_value[1]
+  num_prob <- length(quantile) # 23
+  if (num_prob %% 2 != 1 || num_prob < 3){
+    warning(paste("WIS calculation:",
+                  "Forecaster must return at least 3 symmetric probabilities",
+                  "for this calculation. Returning NA."))
+    return(NA)
```

Comment:
Since this applies to all locations all time periods etc, when we actually do `evaluate_predictions()`, this function may run thousands of times. So if there's one missing value, rather than gumming up everything, better to warn and keep going than stop everything.